notebooks/03-query/10-functions-window.ipynb

```diff
@@ -0,0 +1,241 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "REPLACE INTO \"example_wikipedia_windows\" OVERWRITE ALL\n",
+    "WITH \"ext\" AS (SELECT *\n",
+    "FROM TABLE(\n",
+    "  EXTERN(\n",
+    "    '{\"type\":\"http\",\"uris\":[\"https://druid.apache.org/data/wikipedia.json.gz\"]}',\n",
+    "    '{\"type\":\"json\"}'\n",
+    "  )\n",
+    ") EXTEND (\"timestamp\" VARCHAR, \"channel\" VARCHAR, \"delta\" BIGINT))\n",
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  \"channel\",\n",
+    "  \"delta\"\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "\n",
+    "display.run_task(sql)\n",
+    "sql_client.wait_until_ready('example-wikipedia-windows')\n",
+    "display.table('example-wikipedia-windows')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "622f2158-75c9-4b12-bd8a-c92d30994c1f",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Example window functions queries\n",
+    "\n",
+    "When your data becomes available, you can run window functions queries the data. The following query uses the window function with the SUM aggregation to evaluate the cumulative changes over time within a window.\n",
+    "\n",
+    "The OVER clause defines the window:\n",
+    "- PARTITION BY spefies the dimension that\n",
+    "- ORDER BY orders the rows in the window according to the `__time` column.\n",
+    "\n",
+    "For simplicity's sake, the query limits the query to two channels. You can experiment with adding more channels to the query. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7b71f7ee-4a07-4843-b217-8b4dcd8557f5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query = \"\"\"\n",
+    "SELECT\n",
+    "    FLOOR(__time TO MINUTE) as \"time\",\n",
+    "    channel,\n",
+    "    ABS(delta) AS changes,\n",
+    "    sum(ABS(delta)) OVER (PARTITION BY channel ORDER BY FLOOR(__time TO MINUTE) ASC) AS cum_changes\n",
+    "FROM example_wikipedia_windows\n",
+    "WHERE channel IN ('#kk.wikipedia', '#lt.wikipedia')\n",
+    "GROUP BY channel, __time, delta\n",
+    "\"\"\"\n",
+    "\n",
+    "req = sql_client.sql_request(query)\n",
+    "# Window functions are currently experimental. Set the enableWindiowing\n",
+    "# context parameter to \"true\" to use them.\n",
+    "req.add_context(\"enableWindowing\", \"true\")\n",
+    "display.sql(req)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c21a4b38-3e86-4209-bf47-393b72fc4c7e",
+   "metadata": {},
+   "source": [
+    "The following query illustrates all of the built in window functions using the same data:\n",
+    "\n",
+    "* ROW_NUMBER which returns the numeric value for the row within the window\n",
+    "* RANK which returns the numeric rank of the row within the window\n",
+    "* DENSE_RANK which returns the rank of the row with no gaps. For example, if two rows tie for 1, the next row is ranked 2\n",
```

Comment:
I'd LOVE if this was broken down into additional subheadings, like one on the RANK functions, one on LAG and LEAD, FIRST / LAST. I think it would also help people who are searching for specific functions to find a good example.